src/main/scala/beam/agentsim/infrastructure/ParkingInquiry.scala

```diff
@@ -31,12 +36,65 @@ case class ParkingInquiry(
   requestId: Int = ParkingManagerIdGenerator.nextId, // note, this expects all Agents exist in the same JVM to rely on calling this singleton
   triggerId: Long,
 ) extends HasTriggerId {
-  val activityTypeLowerCased: String = activityType.toLowerCase
 
   def isChargingRequestOrEV: Boolean = {
     beamVehicle match {
       case Some(vehicle) => vehicle.isPHEV || vehicle.isBEV
-      case _             => activityTypeLowerCased == "charge"
+      case _             => activityType == ParkingActivityType.Charge
+    }
+  }
+}
+
+object ParkingInquiry extends LazyLogging {
+  sealed abstract class ParkingActivityType extends EnumEntry
+
+  object ParkingActivityType extends Enum[ParkingActivityType] {
+    val values: immutable.IndexedSeq[ParkingActivityType] = findValues
+
+    case object Charge extends ParkingActivityType
+    case object Init extends ParkingActivityType
+    case object Wherever extends ParkingActivityType
+    case object Home extends ParkingActivityType
+    case object Work extends ParkingActivityType
+    case object Secondary extends ParkingActivityType
+  }
+
+  private def activityTypeStringToEnum(activityType: String): ParkingActivityType = {
+    activityType.toLowerCase match {
+      case "home"      => ParkingActivityType.Home
+      case "init"      => ParkingActivityType.Init
+      case "work"      => ParkingActivityType.Work
+      case "secondary" => ParkingActivityType.Secondary
+      case "charge"    => ParkingActivityType.Charge
+      case "wherever"  => ParkingActivityType.Wherever
+      case otherType =>
+        logger.warn(s"This Parking Activity Type ($otherType) has not been defined")
+        ParkingActivityType.Wherever
     }
   }
+
+  def init(
```

Comment:
You do not need to make this explicit `init` (unless you want to). Instead you can name it `apply` and then all the calls to it no longer need the explicit method name afterwards. It's a bit of Scala magic - and how case classes work in the first place..

Reply:
I used init because apply sometimes it gets me confused. Is it better or it is just matter of taste?

Reply:
Yes, it is preference - I wasn't sure is all then :)